In [1]:
# imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

##### Use the normal cumulative distribution function with the mean and standard deviation of female height to calculate the probability that a female's height is:

1. between 55.0 and 56.0 inches

2. is less than 5 feet (60 in)

3. is greater than the mean (hint: do you know this answer intuitively?)

In [3]:
df = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vRxp-hsw61ck9cpt5ExZIUhvSGT5zYzoUJl_7xr7YF5qZeazC8393q2taqwkDvMxVirgFDJm1WX3uAC/pub?output=csv')
females = df.loc[ df['Gender'] == 'Female'].copy()
females.head()

Gender     Height      Weight
5000  Female  58.910732  102.088326
5001  Female  65.230013  141.305823
5002  Female  63.369004  131.041403
5003  Female  64.479997  128.171511
5004  Female  61.793096  129.781407

In [4]:
# We're only interested in Height data
data = females['Height']

In [6]:
# To answer first question we only want a slice of data between 55 and 56 inches
low = 55
high = 56

In [7]:
# Defining descriptive stats
mean = np.mean(data)
std = np.std(data)
min = np.min(data)
max = np.max(data)

In [11]:
## we subtract to find the probability of a value occurring in the range
stats.norm.cdf(high, loc=mean, scale=std) - stats.norm.cdf(low, loc=mean, scale=std)

0.0015043054289559897

##### There's a .15% chance that a given female is between this specific height

In [13]:
# For a height less that 60 inches
stats.norm.cdf(60, loc=mean, scale=std)

0.08446492725207488

##### That leaves an 8.45% chance of height being less than five feet

##### For final question to probability of height being greater than the mean:
##### It is by necessity 50% of data greater and 50% lesser.